### PR TITLE
[Chassis][Voq][Yang] Make asic_name case sensitive in yang models

### LINKED ISSUE #19485: [Chassis] Incremental config patch fails because the yang validation fails on voq chassis
#### Steps to reproduce the issue:
1. Load the incremental configuration json.patch on the device.
2. Apply the patch
3. The following error is seen
```
Patch Applier: asic1: sorting patch updates.
Failed to apply patch due to: Failed to apply patch on the following scopes:
- asic0: Given patch will produce invalid config. Error: Data Loading Failed
Value "ASIC0" does not satisfy the constraint "[Aa]sic[0-4]" (range, length, or pattern).
- asic1: Given patch will produce invalid config. Error: Data Loading Failed
Value "ASIC0" does not satisfy the constraint "[Aa]sic[0-4]" (range, length, or pattern).
Usage: config apply-patch [OPTIONS] PATCH_FILE_PATH
Try "config apply-patch -h" for help.

```

#### Describe the results you received:
config apply patch failed on chassis linecards
#### Describe the results you expected:
config apply patch should be successful on chassis linecards

#### Output of `show version`:
202405
```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

### PR DESCRIPTION
**Pull request by arlakshm**
#### Why I did it
Fixes #19485 
##### Work item tracking
- Microsoft ADO ** 28529095**:

#### How I did it
Add a new typedef for `asic_name` and make the pattern case in sensitive
#### How to verify it
UT and test on voq linecards

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)
202405 and 202205

- [ ] 
- [ ]

### CODE REVIEW

**Review by xincunli-sonic — APPROVED** [review]
LGTM

File: src/sonic-yang-models/yang-templates/sonic-types.yang.j2 [hunk 1]
```diff
@@ -385,6 +385,11 @@ module sonic-types {
             enum off;
         }
     }
+    typedef asic_name {
```
Comment (xincunli-sonic) [inline src/sonic-yang-models/yang-templates/sonic-types.yang.j2:388]: Nit: add a blank line

Comment (arlakshm) [reply src/sonic-yang-models/yang-templates/sonic-types.yang.j2:389]: fixed it

**Review by abdosi — APPROVED** [review]

**Review by kenneth-arista — APPROVED** [review]

**Comment by gechiang** [comment]
@rlhui , @lguohan , can you help review/merge this PR?

**Pull request merged**